core/src/toga/widgets/canvas/canvas.py

```diff
@@ -0,0 +1,520 @@
+
+    @property
+    def on_press(self) -> OnTouchHandler:
+        """The handler invoked when the canvas is pressed. When a mouse is being used,
+        this press will be with the primary (usually the left) mouse button."""
+        return self._on_press
+
+    @on_press.setter
+    def on_press(self, handler: OnTouchHandler) -> None:
+        self._on_press = wrapped_handler(self, handler)
+
+    @property
+    def on_activate(self) -> OnTouchHandler:
+        """The handler invoked when the canvas is pressed in a way indicating the
+        pressed object should be activated. When a mouse is in use, this will usually be
+        a double click with the primary (usually the left) mouse button.
+
+        This event is not supported on Android or iOS."""
+        return self._on_activate
+
+    @on_activate.setter
+    def on_activate(self, handler: OnTouchHandler) -> None:
+        self._on_activate = wrapped_handler(self, handler)
+
+    @property
+    def on_release(self) -> OnTouchHandler:
+        """The handler invoked when a press on the canvas ends."""
+        return self._on_release
+
+    @on_release.setter
+    def on_release(self, handler: OnTouchHandler) -> None:
+        self._on_release = wrapped_handler(self, handler)
+
+    @property
+    def on_drag(self) -> OnTouchHandler:
+        """The handler invoked when the location of a press changes."""
+        return self._on_drag
+
+    @on_drag.setter
+    def on_drag(self, handler: OnTouchHandler) -> None:
+        self._on_drag = wrapped_handler(self, handler)
+
+    @property
+    def on_alt_press(self) -> OnTouchHandler:
+        """The handler to invoke when the canvas is pressed in an alternate
+        manner. This will usually correspond to a secondary (usually the right) mouse
+        button press.
+
+        This event is not supported on Android or iOS.
+        """
+        return self._on_alt_press
+
+    @on_alt_press.setter
+    def on_alt_press(self, handler: OnTouchHandler) -> None:
+        self._on_alt_press = wrapped_handler(self, handler)
+
+    @property
+    def on_alt_release(self) -> OnTouchHandler:
+        """The handler to invoke when an alternate press is released.
+
+        This event is not supported on Android or iOS.
+        """
+        return self._on_alt_release
+
+    @on_alt_release.setter
+    def on_alt_release(self, handler: OnTouchHandler) -> None:
+        self._on_alt_release = wrapped_handler(self, handler)
+
+    @property
+    def on_alt_drag(self) -> OnTouchHandler:
+        """The handler to invoke when the location of an alternate press changes.
+
+        This event is not supported on Android or iOS.
+        """
+        return self._on_alt_drag
+
+    @on_alt_drag.setter
+    def on_alt_drag(self, handler: OnTouchHandler) -> None:
+        self._on_alt_drag = wrapped_handler(self, handler)
+
+    ###########################################################################
+    # Text measurement
```

Comment:
I don't know the historical background, but are this heading and "As image" below necessary? They each contain only a single method, `measure_text` and `as_image`, respectively.

Reply:
I don't think they're required any more. A long time ago, they were part of some high-level code organization. However, as you've noted, these two are the only two artefacts of that organization that are left, so we might as well get rid of them.